tests/unit/python/foglamp/tasks/purge/test_purge.py

```diff
@@ -0,0 +1,201 @@
+
+        mockStorageClient = MagicMock(spec=StorageClient)
+        mockAuditLogger = AuditLogger(mockStorageClient)
+        with patch.object(FoglampProcess, '__init__'):
+            with patch.object(mockAuditLogger, "__init__", return_value=None):
+                p = Purge(loop=event_loop)
+                p._storage = MagicMock(spec=StorageClient)
+                mock_cm = ConfigurationManager(p._storage)
+                with patch.object(mock_cm, 'get_category_all_items', return_value=mock_cm_get_category()) \
+                        as mock_get_cat:
+                    p.set_configuration()
+                    mock_get_cat.assert_called_once_with('PURGE_READ')
+
+    @pytest.fixture()
+    def store_purge(self, **kwargs):
+        if kwargs.get('age') == '-1' or kwargs.get('size') == '-1':
+            return {"message": "409 Conflict"}
+        return {"readings": 10, "removed": 1, "unsentPurged": 2, "unsentRetained": 7}
+
+    config = [{"retainUnsent": {"value": "False"}, "age": {"value": "72"}, "size": {"value": "20"}},
+              {"retainUnsent": {"value": "False"}, "age": {"value": "72"}, "size": {"value": "0"}},
+              {"retainUnsent": {"value": "False"}, "age": {"value": "0"}, "size": {"value": "100"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "72"}, "size": {"value": "20"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "72"}, "size": {"value": "0"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "0"}, "size": {"value": "100"}},
+              {"retainUnsent": {"value": "False"}, "age": {"value": "0"}, "size": {"value": "0"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "0"}, "size": {"value": "0"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "-1"}, "size": {"value": "-1"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "bla"}, "size": {"value": "0"}},
+              {"retainUnsent": {"value": "True"}, "age": {"value": "0"}, "size": {"value": "bla"}}]
+
+    @pytest.mark.parametrize("conf, expected_return, expected_calls, valid_purge, valid_store_return", [
+        (config[0], (2, 4), {'sent_id': 0, 'size': '20', 'flag': 'purge'}, True, True),
+        (config[1], (1, 2), {'sent_id': 0, 'age': '72', 'flag': 'purge'}, True, True),
+        (config[2], (1, 2), {'sent_id': 0, 'size': '100', 'flag': 'purge'}, True, True),
+        (config[3], (2, 4), {'sent_id': 0, 'size': '20', 'flag': 'retain'}, True, True),
+        (config[4], (1, 2), {'sent_id': 0, 'age': '72', 'flag': 'retain'}, True, True),
+        (config[5], (1, 2), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, True, True),
+        (config[6], (0, 0), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, False, True),
+        (config[7], (0, 0), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, False, True),
+        (config[8], (0, 0), {'sent_id': 0, 'size': '100', 'flag': 'retain'}, True, False)
```

Comment:
instead of 0, 8 index, can we have a meaningful key? what test_input we want to use config[purpose_key]